File: dashviagens/src/main/java/com/dashviagens/cost/dto/BudgetResponse.java
package com.dashviagens.cost.dto;

import java.math.BigDecimal;

public record BudgetResponse(
        String countryCode,
        int days,
        BigDecimal estimatedFlight,
        BigDecimal estimatedHotel,
        BigDecimal estimatedFood,
        BigDecimal estimatedTransport,
        BigDecimal estimatedActivities,
        BigDecimal totalEstimated,
        BigDecimal totalBudget,
        BigDecimal remaining,
        boolean withinBudget
) {
}

File: dashviagens/frontend/src/types/BudgetResponse.ts
export interface BudgetResponse {
  countryCode: string;
  days: number;
  estimatedFlight: number;
  estimatedHotel: number;
  estimatedFood: number;
  estimatedTransport: number;
  estimatedActivities: number;
  totalEstimated: number;
  totalBudget: number;
  remaining: number;
  withinBudget: boolean;
}